frontend/ui/wikipedia.lua

```diff
@@ -33,9 +33,24 @@ function Wikipedia:loadPage(text, lang, intro, plain)
     local http = require('socket.http')
     local https = require('ssl.https')
     local ltn12 = require('ltn12')
-
+    local _ = require("gettext")
+    local DataStorage = require("datastorage")
+    local DocSettings = require("docsettings")
+    --read language from current book
+    local lastfile = G_reader_settings:readSetting("lastfile")
+    local book_stats = DocSettings:open(lastfile):readSetting('stats')
```

Comment:
I think the proper place to set the lang is in `ReaderWikipedia:onLookupWikipedia` method. You can set it in the following order:

1. check book language setting
2. check reader language setting
3. use Translator.detect to guess the language from the word

Yon can access the book language property inside `ReaderWikipedia`  by `self.view.document:getProps().language`. `G_reader_settings` is already a global variable so it's accessible everywhere, you don't need to reopen the setting file ;)